enclave-tls/src/tls_wrappers/openssl/un_negotiate.c

```diff
@@ -301,5 +389,12 @@ int verify_certificate(int preverify, X509_STORE_CTX *ctx)
 		return 0;
 	}
 
+#ifdef SGX
+	err = enclave_user_callback(tls_ctx, &evidence);
+	if (err != TLS_WRAPPER_ERR_NONE) {
+		ETLS_ERR("failed to verify user enclave info\n");
+		return 0;
+	}
+#endif
```

Comment:
Allowing to call a user_callback is a common mechanism, so should not be only available for sgx build.